coordinator/internal/transitengine/crypto.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package transitengine
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/edgelesssys/contrast/internal/crypto"
+
+	b64 "encoding/base64"
+)
+
+// symmetricOpts holds parameters related to the performed symmetric encryption, specifyable as http request parameters.
+type symmetricOpts struct {
+	// Convergent TODO(jmxzo): add parameter support
+	convergent bool
+	// Nonce
+	nonce []byte
+	// AdditionalData
+	additionalData []byte
+}
+
+type encryptionMap struct {
+	Plaintext     string `json:"plaintext"`
+	Ciphertext    string `json:"ciphertext"`
+	symmetricOpts symmetricOpts
+}
+
+// Note: In case only err is returned the resulting httpError will fallbback to InternalServerError status code 500.
+
+func handleEncrypt(w http.ResponseWriter, r *http.Request, derivedTransitKey []byte) error {
+	encryptionMapping, err := parseEncryptionRequest(r)
+	if err != nil {
+		return err
+	}
+	plaintext, err := b64.StdEncoding.DecodeString(encryptionMapping.Plaintext)
+	if err != nil {
+		return httpError{error: err, status: http.StatusBadRequest}
+	}
+
+	ciphertextBytes, err := symmetricEnc(derivedTransitKey, plaintext, encryptionMapping.symmetricOpts)
+	if err != nil {
+		return err
+	}
+	encryptionMapping.Ciphertext = "vault:v1:" + b64.StdEncoding.EncodeToString(ciphertextBytes)
+	if err = writeJSONResponse(w, http.StatusOK, createRespBody("ciphertext", encryptionMapping.Ciphertext)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func handleDecrypt(w http.ResponseWriter, r *http.Request, derivedTransitKey []byte) error {
+	encryptionMapping, err := parseEncryptionRequest(r)
+	if err != nil {
+		return err
+	}
+	ciphertext := strings.TrimPrefix(encryptionMapping.Ciphertext, "vault:v1:")
+	ciphertextBytes, err := b64.StdEncoding.DecodeString(ciphertext)
+	if err != nil {
+		return httpError{error: err, status: http.StatusBadRequest}
+	}
+	encryptionMapping.symmetricOpts.nonce = ciphertextBytes[:12]
+
+	plaintextBytes, err := symmetricDec(derivedTransitKey, ciphertextBytes[12:], encryptionMapping.symmetricOpts)
+	if err != nil {
+		return err
+	}
+	encryptionMapping.Plaintext = b64.StdEncoding.EncodeToString(plaintextBytes)
+	if err = writeJSONResponse(w, http.StatusOK, createRespBody("plaintext", encryptionMapping.Plaintext)); err != nil {
+		return err
+	}
+	return nil
+}
+
+// symmetricEnc returns the encrypted plaintext based on the symmetric options and encryption key handed in.
+func symmetricEnc(encKey, plaintext []byte, _ symmetricOpts) ([]byte, error) {
+	aesCipher, err := aes.NewCipher(encKey)
+	if err != nil {
+		return nil, err
+	}
+	gcm, err := cipher.NewGCM(aesCipher)
```

Comment:
Using the raw Go crypto libs feels a bit sharp. I think that tink doesn't have good support for this use-case. Have we checked that there isn't a good library for this?